### PR TITLE
Document Deferred Credential endpoint

### LINKED ISSUE #17: Define Token Request error responses
Originally ideated upon here https://github.com/TBD54566975/known-customer-credential/pull/12

We need to support all of the standard OAuth error responses https://datatracker.ietf.org/doc/html/rfc6749#section-4.2.2.1

As well as a few custom to OID4VCI https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-token-error-response

**_Specifically_** we need to add support to `authorization_pending` which will be returned in the event the IDV case is pending status

### LINKED ISSUE #19: Define deferred credential issuance flow & endpoints
We don't currently have anything defined for the deferred credential flow, other than the existence of the `transaction_id` in the Credential Response

### LINKED ISSUE #44: Remove references to `authorization_pending`
The Token endpoint response code `authorization_pending` has been removed from OID4VCI, so we should remove it too. https://github.com/openid/OpenID4VCI/pull/314

### PR DESCRIPTION
**Pull request by tomdaffurn**
Closes #44 
Closes #19 
Closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -561,9 +573,30 @@ The `access_token` granted by the Credential Issuer contains the following [JWT
...
+- `unauthorized_client`
+- `unsupported_grant_type`
+
+### Credential Endpoint
+
+This endpoint issues a Credential (or a Transaction ID) upon presentation of a valid Access Token.
+The URL is the `credential_endpoint` field of the [Credential Issuer Metadata](#credential-issuer-metadata).
 
 ```mermaid
 sequenceDiagram
```
Comment (KendallWeihe) [inline README.md:602]: Thoughts on adding [an `alt`/`else` block](https://mermaid.js.org/syntax/sequenceDiagram.html#alt) for sync vs async flows? My general thinking is... we want to convey that the _Credential Endpoint_ is hit first, but it doesn't necessitate the _Deferred Credential Flow_.

File: README.md [hunk 2]
```diff
@@ -621,32 +651,67 @@ For the client to prove possession of their cryptographic materials, they must c
...
+
+##### Authorization Errors
+
+If the Credential Request does not contain a valid Access Token, the response is an authorization error response, as per [RFC6750](https://www.rfc-editor.org/rfc/rfc6750.html#section-3).
+
+#### Credential Request Errors
+
+| Field               | Description                               | Required | References                                                                                                 | Comments |
+| :------------------ | :---------------------------------------- | :------- | :--------------------------------------------------------------------------------------------------------- | :------- |
+| `error`             | A string error code from below list       | y        | [OID4VCI](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-7.3.1.2-3.1.1) |          |
```
Comment (KendallWeihe) [inline README.md:684]: Might be good to list them out below (as well as the link you've added) like you did in the _Token Error Response_ section above

**Review by decentralgabe — APPROVED** [review]
lgtm, have cross-checked with the spec and not found any issues

File: README.md (hunk 1 above)
Comment (tomdaffurn) [reply README.md:602]: Niiice, done 👍

File: README.md (hunk 2 above)
Comment (tomdaffurn) [reply README.md:688]: yep, done

**Pull request merged**